rpcs3/Emu/RSX/VK/VKGSRender.cpp

```diff
@@ -3373,6 +3373,46 @@ void VKGSRender::flip(int buffer, bool emu_flip)
 		vk::change_image_layout(*m_current_command_buffer, target_image, VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL, present_layout, range);
 	}
 
+	if (m_frame->screenshot_toggle == true)
+	{
+		m_frame->screenshot_toggle = false;
+
+		const size_t sshot_size = buffer_height * buffer_width * 4;
+
+		const auto device = vk::get_current_renderer();
+		vk::buffer sshot_vkbuf(
+		    *device, 256 * 0x100000, device->get_memory_mapping().host_visible_coherent, VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT | VK_MEMORY_PROPERTY_HOST_COHERENT_BIT, VK_BUFFER_USAGE_TRANSFER_SRC_BIT, 0);
+
+		flush_command_queue(true);
```

Comment:
Since you're supplying a dedicated buffer for this allocation, this flush is not necessary.